kivakit-core/src/main/java/com/telenav/kivakit/core/thread/KivaKitThread.java

```diff
@@ -422,16 +426,16 @@ public Time startedAt()
     /**
      * @return The state that this thread is in
      */
-    public StateMachine<State> state()
+    public StateMachine<State> stateMachine()
     {
-        return state;
+        return stateMachine;
     }
 
     /**
      * Attempt to stop this thread, waiting for the maximum specified time for it to exit
      */
     @Override
-    public void stop(LengthOfTime maximumWait)
+    public void stop(Duration maximumWait)
```

Comment:
Given that `Thread.stop()` is dead in the JDK, do we want this?

Reply:
it doesn't work that way. it signals via a StateMachine and waits for the thread to exit:

```
        whileLocked(() ->
        {
            if (!is(EXITED))
            {
                transition(STOP_REQUESTED);
                interrupt();
                waitFor(EXITED, maximumWait);
            }
        });
    }
```

Reply:
Have a look. Should be resolved now.